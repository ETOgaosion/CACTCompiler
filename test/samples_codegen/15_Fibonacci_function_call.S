iobuff:
	.zero 80
.text
.align 1
.globl	fibonacci
.type	fibonacci, @function
fibonacci:
	sd	ra, -8(sp)
	sw	a0, -12(sp)
	mv	t0, zero
	sw	t0, -28(sp)
	addi	t0, zero, 1
	sw	t0, -24(sp)
	mv	t0, zero
	sw	t0, -20(sp)
	mv	t0, zero
	sw	t0, -16(sp)
L_1_fibonacci:
	lw	t0, -20(sp)
	lw	t1, -12(sp)
	slt	t2, t0, t1
	sb	t2, -36(sp)
	lb	t0, -36(sp)
	beqz	t0, L_0_fibonacci
	lw	t1, -28(sp)
	lw	t2, -24(sp)
	addw	t0, t1, t2
	sw	t0, -40(sp)
	lw	t0, -40(sp)
	sw	t0, -16(sp)
	lw	a0, -16(sp)
	addi	sp, sp, -40
	call	print_int
	addi	sp, sp, 40
	lw	t0, -24(sp)
	sw	t0, -28(sp)
	lw	t0, -16(sp)
	sw	t0, -24(sp)
	lw	t1, -20(sp)
	addi	t0, t1, 1
	sw	t0, -32(sp)
	lw	t0, -32(sp)
	sw	t0, -20(sp)
	lla	t0, L_1_fibonacci
	jr	t0
L_0_fibonacci:
	lw	a0, -16(sp)
	ld	ra, -8(sp)
	ret
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	mv	t0, zero
	sw	t0, -12(sp)
	addi	sp, sp, -24
	call	get_int
	addi	sp, sp, 24
	sw	a0, -20(sp)
	lw	t0, -20(sp)
	sw	t0, -12(sp)
	lw	a0, -12(sp)
	addi	sp, sp, -24
	call	fibonacci
	addi	sp, sp, 24
	sw	a0, -16(sp)
	mv	a0, zero
	ld	ra, -8(sp)
	ret
