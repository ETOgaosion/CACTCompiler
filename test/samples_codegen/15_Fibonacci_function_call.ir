============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
fibonacci(2) => 2:
----------- paramVars -----------
n := symbol type: 0; data type: 2
----------- localVars -----------
c := symbol type: 1; data type: 2; initValue: 0

i := symbol type: 1; data type: 2; initValue: 0

b := symbol type: 1; data type: 2; initValue: 1

a := symbol type: 1; data type: 2; initValue: 0

----------- tempVars ------------
t_5:= symbol type: TEMP VAR; data type: 2
t_0:= symbol type: TEMP VAR; data type: 1
t_1:= symbol type: TEMP VAR; data type: 2
----------- Codes ------------
fibonacci:
	n =  param 0;
	a = 0;
	b = 1;
	i = 0;
	c = 0;
L_1_fibonacci:
	t_0 = i < n
	beqz t_0 L_0_fibonacci
	t_1 = a + b;
	t_2 <= c;
	t_2 = t_1;
	addparam c;
	call print_int, frame size: 40; 
	t_3 <= a;
	t_3 = b;
	t_4 <= b;
	t_4 = c;
	t_5 = i + 1;
	t_6 <= i;
	t_6 = t_5;
	goto L_1_fibonacci
L_0_fibonacci:
	return ;

----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
n := symbol type: 1; data type: 2; initValue: 0

----------- tempVars ------------
t_2:= symbol type: TEMP VAR; data type: 2
t_0:= symbol type: TEMP VAR; data type: 2
----------- Codes ------------
main:
	n = 0;
	call get_int, frame size: 24; 
	t_0 =  return ;
	t_1 <= n;
	t_1 = t_0;
	addparam n;
	call fibonacci, frame size: 24; 
	t_2 =  return ;
	return 0;

