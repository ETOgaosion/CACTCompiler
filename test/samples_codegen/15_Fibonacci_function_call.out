bbl loader
please enter an int number:
1
2
3
5
8
13
21
34
55
89
